test/watchos_framework_test.sh

```diff
@@ -0,0 +1,1946 @@
+
+  assert_equals "my.bundle.id" \
+      "$(cat "test-bin/app/CFBundleIdentifier_app")"
+
+  assert_equals "CommonValue" \
+      "$(cat "test-bin/app/CommonKey_app")"
+
+  do_build watchos //app:dump_plist_framework || fail "Should build"
+
+  assert_equals "my.framework.id" \
+      "$(cat "test-bin/app/CFBundleIdentifier_framework")"
+  assert_equals "CommonValue" \
+      "$(cat "test-bin/app/CommonKey_framework")"
+
+  do_build watchos //app:dump_plist_depframework || fail "Should build"
+
+  assert_equals "my.depframework.id" \
+      "$(cat "test-bin/app/CFBundleIdentifier_depframework")"
+  assert_equals "CommonValue" \
+      "$(cat "test-bin/app/CommonKey_depframework")"
+}
+
+# Verifies that, when an extension depends on a framework with different
+# minimum_os, symbol subtraction still occurs.
+function test_differing_minimum_os() {
+  create_minimal_watchos_framework_with_params True "2.0"
+
+  cat > app/BUILD <<EOF
+load("@build_bazel_rules_apple//apple:watchos.bzl",
+     "watchos_application",
+     "watchos_extension",
+)
+
+objc_library(
+    name = "lib",
+    srcs = ["main.m"],
+    deps = ["//framework:framework_lib"],
+)
+
+watchos_application(
+    name = "app",
+    bundle_id = "my.bundle.id",
+    extension = ":ext",
+    infoplists = ["Info-App.plist"],
+    minimum_os_version = "2.0",
+    provisioning_profile = "@build_bazel_rules_apple//test/testdata/provisioning:integration_testing_ios.mobileprovision",
+    # deps = [":lib"],
+)
+
+watchos_extension(
+    name = "ext",
+    bundle_id = "my.bundle.id.extension",
+    frameworks = ["//framework:framework"],
+    infoplists = ["Info-Ext.plist"],
+    minimum_os_version = "3.0",
+    provisioning_profile = "@build_bazel_rules_apple//test/testdata/provisioning:integration_testing_ios.mobileprovision",
+    deps = [":lib"],
+)
+EOF
+
+  cat > app/main.m <<EOF
+#import <Foundation/Foundation.h>
+// This dummy class is needed to generate code in the extension target,
+// which does not take main() from here, rather from an SDK.
+@interface Foo: NSObject
+@end
+@implementation Foo
+@end
+
+int main(int argc, char **argv) {
+  return 0;
+}
+EOF
+
+  cat > app/Info-App.plist <<EOF
+{
+  CFBundleIdentifier = "\${PRODUCT_BUNDLE_IDENTIFIER}";
+  CFBundleName = "\${PRODUCT_NAME}";
+  CFBundlePackageType = "APPL";
+  CFBundleShortVersionString = "1.0";
+  CFBundleVersion = "1.0";
+}
+EOF
+
+  cat > app/Info-Ext.plist <<EOF
+{
+  CFBundleName = "\${PRODUCT_NAME}";
+  CFBundlePackageType = "APPL";
+  CFBundleShortVersionString = "1.0";
+  CFBundleVersion = "1.0";
+  NSExtension = {
+    NSExtensionAttributes = {
+      WKAppBundleIdentifier = "my.bundle.id";
+      NSExtensionPointIdentifier = com.apple.watchkit;
+    };
+    WKExtensionDelegateClassName = "DummyValue";
+  };
+}
+EOF
+
+  do_build watchos //app:app || fail "Should build"
+
+  assert_binary_not_contains watchos "test-bin/app/app.zip" \
+      "app.app/PlugIns/ext.appex/ext" "doStuff"
```

Comment:
This is one of the tests failing:
```
test_differing_minimum_os FAILED: Expected binary 'tempdir/fat_binary' to not contain 'doStuff' but it did. contents were: tempdir/fat_binary:
SYMBOL TABLE:
0000bef8 l     F __TEXT,__stub_helper  stub helpers
0000c024 l     O __DATA,__objc_classlist _OBJC_LABEL_CLASS_$
0000c030 l     O __DATA,__objc_const __OBJC_METACLASS_RO_$_Foo
0000c058 l     O __DATA,__objc_const __OBJC_CLASS_RO_$_Foo
0000c0a8 l     O __DATA,__data __dyld_private
0000c094 g     O __DATA,__objc_data _OBJC_CLASS_$_Foo
0000c080 g     O __DATA,__objc_data _OBJC_METACLASS_$_Foo
0000be64 g     F __TEXT,__text _WKExtensionMain
00004000 g     F __TEXT,__text __mh_execute_header
0000be48 g     F __TEXT,__text _doStuff
0000be60 g     F __TEXT,__text _main .
``` 
I don't understand why we don't expect this symbol to be in the extension. This test was inspired by https://github.com/bazelbuild/rules_apple/blob/193d258479e891bf286bf884d52ac779c83be206/test/tvos_framework_test.sh#L1493 which is similar, but even there I set the minimum os to be the same in the framework and app, and the test passed as well. So maybe this is not relevant anymore?